apps/api/src/app/content-templates/usecases/parse-preview/preview-email.usecase.ts

```diff
@@ -0,0 +1,57 @@
+import { Injectable } from '@nestjs/common';
+import { IEmailBlock, OrganizationRepository, OrganizationEntity } from '@novu/dal';
+import { CompileTemplate } from '../compile-template/compile-template.usecase';
+import { CompileTemplateCommand } from '../compile-template/compile-template.command';
+import { PreviewEmailCommand } from './preview-email.command';
+
+@Injectable()
+export class PreviewEmail {
+  constructor(private compileTemplate: CompileTemplate, private organizationRepository: OrganizationRepository) {}
+
+  public async execute(command: PreviewEmailCommand) {
+    const organization: OrganizationEntity = await this.organizationRepository.findById(command.organizationId);
+    const isEditorMode = command.contentType === 'editor';
+    const content = await this.getContent(isEditorMode, command.content);
+
+    const html = await this.compileTemplate.execute(
+      CompileTemplateCommand.create({
+        templateId: isEditorMode ? 'basic' : 'custom',
+        customTemplate: !isEditorMode ? (content as string) : undefined,
+        data: {
+          blocks: isEditorMode ? content : [],
+          branding: {
+            logo: organization.branding?.logo,
+            color: organization.branding?.color || '#f47373',
+          },
+        },
+      })
+    );
+
+    return { html };
+  }
+
+  private async getContent(isEditorMode, content: string | IEmailBlock[]): Promise<string | IEmailBlock[]> {
+    if (isEditorMode) {
+      content = [...content] as IEmailBlock[];
```

Comment:
I do not get very well what's happening here. If you send a string and you spread it, the array is going to be made of individual characters.
How does the array spread distinguish the case between a string and an IEmailBlock? I honestly would expect the string case to not work properly in the next par of the code.